### DISCUSSION TITLE
Is there a way to use third party geocodes? If so is there any free ones for Australia?

### DISCUSSION THREAD

**Discussion by Raj2032** [organicmaps/organicmaps#3492] [Q&A]
First of all is it possible for me to add third party geocodes to organic maps?

If this is true then is it possible for me to use a geocode third party service for free which supports Australia?

This is because everytime I try to put in the exact address, organic maps cannot find this and it is unfortunate but I have to still use Google Maps because Organic Maps can't find exact addresses. And I want to use organic maps but yeah just wondering?

**Comment by Outlet2048** [comment]
Have you checked if the locations are mapped on OpenStreetMap.org?

**Reply by Raj2032** [reply]
Yes and they are unmappable

**Reply by TheAdventurer64** [reply]
Why can't they be mapped?

**Reply by Raj2032** [reply]
You type in "23 Harold street, Springfield" and it will ONLY show "Harold Street, Springfield", it doesn't have street addresses.

**Reply by Outlet2048** [reply]
Are you able to find those addresses when you zoom into the map on openstreetmap.org?

**Reply by biodranik** [reply]
The main point here is that Organic Maps can find only those addresses that are added/present on OpenStreetMap.org.

**Reply by Raj2032** [reply]
@AlertSubject Nope not at all mate. Just the street name that's about it!

**Reply by Raj2032** [reply]
@biodranik That is why I am saying that I wish there was a way to add third party geocodes on organic maps, I am not sure how MagicEarth that uses openstreetmaps is able to have all geocodes?

**Reply by biodranik** [reply]
They either take some free public data or buy this data/pay per request to some provider. Are you aware of any publicly available source of Australia addresses?

**Reply by Outlet2048** [reply]
Magic earth uses another source for addresses

Perhaps you could add the missing addresses on OpenStreetMap.org so that they will be included in the next mal update

**Reply by Raj2032** [reply]
Maybe yeah

**Comment by biodranik** [comment]
At the moment we're not ready to support online geocoders, maybe later we consider it.

But it is possible in theory to integrate some open source address databases to work in offline.

**Reply by Raj2032** [reply]
Ah right, thanks
